internal/tofu/transform_provider.go

```diff
@@ -245,6 +247,127 @@ func (t *ProviderTransformer) Transform(g *Graph) error {
 	return diags.Err()
 }
 
+// ProviderFunctionTransformer is a GraphTransformer that maps resources to providers
+// within the graph. This will error if there are any resources that don't map
```

Comment:
I would add that it only does so for resources that use functions from the given provider.